### PR TITLE
Make root plugin class

### PR DESCRIPTION
**Pull request by wswoodruff**
This PR might be controversial in that I decided a root `internals.Vision` class might be handy and I feel like these changes will help with future plans I have

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -15,151 +15,184 @@ const Manager = require('./manager');
...
+                }
+
+                return h.view(settings.template, context, settings.options);
+            };
+        };
+    }
 
 
-internals.state = (realm) => {
+    getState(realm) {
```
Comment (devinivy) [inline lib/index.js]: I find it a little funny having instance methods that don't utilize `this`.  Any reason this helper is moved out of `internals`?

File: lib/index.js [hunk 2]
```diff
@@ -15,151 +15,184 @@ const Manager = require('./manager');
...
     pkg: require('../package.json'),
 
     register: function (server, options) {
 
+        return new internals.vision(server, options);
+    }
+};
+
+
+internals.vision = class Vision {
```
Comment (devinivy) [inline lib/index.js]: Would expect this to be `internals.Vision`.

File: lib/index.js [hunk 3]
```diff
@@ -15,151 +15,184 @@ const Manager = require('./manager');
...
+            return manager;
+        };
     }
 
-    const manager = new Manager(options);
-    realmState.manager = manager;
-    return manager;
-};
 
+    render(rootServer) {
```
Comment (devinivy) [inline lib/index.js]: Are these really root servers?  It seems like the plugin's server is passed in.

**Comment by wswoodruff** [comment]
Reverting these changes because it's not useful, smells too much like making a class just to make a class

**Pull request closed without merging**